packages/pymedphys_dicom/src/pymedphys_dicom/dicom/_level2/anonymise.py

```diff
@@ -109,12 +110,24 @@
                                        'LO': "Anonymous",
                                        'PN': "Anonymous",
                                        'SH': "Anonymous",
-                                       'SQ': ["Anonymous"],
+                                       'SQ': [Dataset()],
                                        'ST': "Anonymous",
                                        'TM': "000900.000000",
                                        'UI': "12345678"}
 
 
+def anonymised_dicom_filepath(filepath, ds=None, preserve_original=False):
+    if preserve_original:
+        basename_anon = "{}_Anonymised.dcm".format('.'.join(
+            basename(filepath).split('.')[:-1]))
+    else:
+        mode_prefix = DICOM_SOP_CLASS_NAMES_MODE_PREFIXES[ds.SOPClassUID.name]
```

Comment:
Will `ds.SOPClassUID` fail here if no `ds` is passed?

Reply:
Ahh, I see what's going on here. I think this is clearly two functions squished awkwardly into one. For example, one of them would be:

```python
def create_filename_from_dataset(dataset):
    mode_prefix = DICOM_SOP_CLASS_NAMES_MODE_PREFIXES[ds.SOPClassUID.name]
    filename = "{}.{}".format(mode_prefix, ds.SOPInstanceUID)

	return filename
```

Reply:
See commit c9767ad :)